interface/forms/misc_billing_options/new.php

```diff
@@ -20,15 +22,23 @@
 
 use OpenEMR\Core\Header;
 
-if (! $encounter) { // comes from globals.php
+if (isset($_REQUEST['isBilling'])) {
+    $pid = $_SESSION['billpid'] = attr($_REQUEST['pid']);
+    $encounter = $_SESSION['billencounter'] = attr($_REQUEST['enc']);
+} elseif (isset($_SESSION['billencounter'])) {
+    unset($_SESSION['billpid']);
+    unset($_SESSION['billencounter']);
+}
+
+if (!$encounter) { // comes from globals.php
     die(xlt("Internal error: we do not seem to be in an encounter!"));
 }
-//only one misc billing form so grab if exists
+//only one misc billing form per encounter so grab if exists
 $formid = 0 + (isset($_GET['id']) ? $_GET['id'] : 0);
 if (empty($formid)) {
     $mboquery = sqlquery("SELECT `fmbo`.`id` FROM `form_misc_billing_options` AS `fmbo`
                           INNER JOIN `forms` ON (`fmbo`.`id` = `forms`.`form_id`) WHERE
-                          `forms`.`deleted` = 0 AND `forms`.`form_name` = 'Misc Billing Options' AND
+                          `forms`.`deleted` = 0 AND `forms`.`formdir` = 'misc_billing_options' AND
```

Comment:
very smart change :)

Reply:
I do still have fleeting moments of clarity.

Reply:
yep, I remember having those in the past. It's been a couple years for me :)